### DISCUSSION TITLE
ds18b20 connection

### DISCUSSION THREAD

**Discussion by espenfjo** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#628] [Q&A] [answered]
Hi

Just to verify. 

To connect a DS18B20 for ambient temperature I have to connect ground to ground, VIN to 3.3V, and output to a free digital pin, and then use a 4.7K resistor between output and 3.3V?

**Comment by visualapproach** [comment, marked as answer]
Yes, but not D0 or D8

**Comment by lukefoster197** [comment]
Sorry can I just ask. I have connected my temp sensor as above but without the resistor. Does the resistor go between D0 and 3.3v or not? 
I have the output connected directly to D0 at the moment.

**Reply by pH03NIxM** [reply]
You need 4.7k pullup resistor between the digital output and 3V, othercase the sensor is not working. On 4 pin pump with V2 PCB i used D7 and working fine (but only with the resistor).
